core/src/test/java/bio/terra/pearl/core/service/workflow/EnrollmentWorkflowTests.java

```diff
@@ -249,6 +251,35 @@ public void testProxyEnrollingMultipleChild(TestInfo testInfo){
         assertThat(participantTaskService.findByEnrolleeId(governedEnrollee2.getId()), hasSize(1));
     }
 
+    @Test
+    @Transactional
+    public void testDetectingProxyWhileEnrollment(TestInfo info) {
+        PortalEnvironment portalEnv = portalEnvironmentFactory.buildPersisted(getTestName(info));
+        StudyEnvironment studyEnv = studyEnvironmentFactory.buildPersisted(portalEnv, getTestName(info));
+        Survey preEnrollmentSurvey = surveyFactory.buildPersisted(getTestName(info));
+        StudyEnvironmentSurvey.builder()
+                .surveyId(preEnrollmentSurvey.getId())
+                .studyEnvironmentId(studyEnv.getId())
+                .build();
+        String proxyQuestionStableId = "proxyQuestion";
+        preEnrollmentSurveyFactory.buildPersistedProxyAnswerMapping(getTestName(info), preEnrollmentSurvey.getId(), proxyQuestionStableId);
+        String preEnrollmentSurveyResponseForProxy = """
+                [{"questionStableId":"proxyQuestion","surveyVersion":0,"viewedLanguage":"en","stringValue":"true"},
+                {"createdAt":1710527339.202811000,"lastUpdatedAt":1710527339.202811000,"questionStableId":"qualified","surveyVersion":0,"booleanValue":true}]""";
+        ParticipantUserFactory.ParticipantUserAndPortalUser userProxyBundle = participantUserFactory.buildPersisted(portalEnv, getTestName(info));
+        PreEnrollmentResponse preEnrollmentResponseProxy = preEnrollmentSurveyFactory.buildPersisted(getTestName(info), preEnrollmentSurvey.getId(), true, preEnrollmentSurveyResponseForProxy, userProxyBundle.ppUser().getId(), userProxyBundle.user().getId(), studyEnv.getId());
+        Assertions.assertTrue(enrollmentService.isProxyEnrollment(preEnrollmentResponseProxy.getId()));
+
+        String preEnrollmentSurveyResponseForRegularUser = """
+                [{"questionStableId": "proxyQuestion","surveyVersion":0,"viewedLanguage":"en","stringValue":"false"},
+                {"createdAt":1710527339.202811000,"lastUpdatedAt":1710527339.202811000,"questionStableId":"qualified","surveyVersion":0,"booleanValue":true}]""";
+        ParticipantUserFactory.ParticipantUserAndPortalUser userBundle = participantUserFactory.buildPersisted(portalEnv, getTestName(info));
+        PreEnrollmentResponse preEnrollmentResponse = preEnrollmentSurveyFactory.buildPersisted(getTestName(info), preEnrollmentSurvey.getId(), true, preEnrollmentSurveyResponseForRegularUser, userBundle.ppUser().getId(), userBundle.user().getId(), studyEnv.getId());
+        preEnrollmentResponseProxy.setQualified(true);
+        Assertions.assertFalse(enrollmentService.isProxyEnrollment(preEnrollmentResponse.getId()));
+
```

Comment:
this is great!  Now add one more check that confirms that we actually do a proxy enroll based on the results.

So, e.g.
```
  HubResponse<Enrollee> hubResponse = enrollmentService.enroll(studyEnv.getEnvironmentName(), studyShortcode,
                userProxyBundle.user(), userProxyBundle.ppUser(), preEnrollmentResponseProxy.getId());
// confirm that two enrollees were created, and only one is a subject  
assertThat(hubResponse.enrollee().isSubject(), equalTo(false));
assertThat(hubResponse.response().isSubject(), equalTo(true));

Reply:
Done ✅